File: newDemo/newDemo/Base/Photo/AddPhotoCollectionViewCell.m
//
//  AddPhotoCollectionViewCell.m
//  BaseToolSystem
//
//  Created by 靳建南 on 2019/4/23.
//  Copyright © 2019年 靳建南. All rights reserved.
//

#import "AddPhotoCollectionViewCell.h"
#define  photoWeight (ScreenW-50)/4 //CollectionView的宽
@implementation AddPhotoCollectionViewCell

-(id)initWithFrame:(CGRect)frame{
    if (self = [super initWithFrame:frame]) {
        self.backgroundColor = [UIColor whiteColor];
        
        //添加照片
        _addImgBtn = [UIButton btnWithFrame:CGRectMake(0,0, photoWeight, photoWeight) title:@"请上传图片" titleColor:LAB3_COLOR titleFont:10 image:@"默认图片"];
        [_addImgBtn btnLayoutWithEdgeInsetsStyle:MKButtonEdgeInsetsStyleTop imageTitleSpace:8];
        _addImgBtn.userInteractionEnabled = NO;
        
        //给按钮绘虚线
        CAShapeLayer *border = [CAShapeLayer layer];
        border.strokeColor = LAB3_COLOR.CGColor;//虚线颜色
        border.fillColor = nil;
        border.path = [UIBezierPath bezierPathWithRect:_addImgBtn.bounds].CGPath;
        border.frame = _addImgBtn.bounds;
        border.lineWidth = 1.f;
        border.lineCap = @"square";
        border.lineDashPattern = @[@2, @2];
        [_addImgBtn.layer addSublayer:border];
        
        [self.contentView addSubview:_addImgBtn];
    }
    return self;
}

@end

File: BaseToolClass/BaseToolClass/Base(基类)/PhotoAddASeek(图片添加及展示)/AddPhotoCollectionViewCell.swift
//
//  AddPhotoCollectionViewCell.swift
//  BaseToolClass
//
//  Created by 靳建南 on 2018/11/9.
//  Copyright © 2018年 靳建南. All rights reserved.
//

import UIKit

class AddPhotoCollectionViewCell: BaseCollectionViewCell {
    
    var addImgBtn:UIButton!
    let photoWeight = (ScreenW-50)/4 //CollectionView的宽
    
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        self.backgroundColor = UIColor.white
        
        //添加照片
        addImgBtn = UIButton.btn(btnWithFrame: CGRect.init(x: 0, y: 0, width: photoWeight, height: photoWeight), title: "添加照片", titleColor: LAB3_COLOR, titleFont: 10, image: "相机")
        addImgBtn.btnLayoutWithEdgeInsetsStyle(style: .MKButtonEdgeInsetsStyleTop, space: 8)
        addImgBtn.isUserInteractionEnabled = false
        
        //给按钮绘虚线
        let border:CAShapeLayer = CAShapeLayer()
        border.strokeColor = LAB3_COLOR.cgColor //虚线颜色
        border.fillColor = nil
        border.path = UIBezierPath.init(rect: addImgBtn.bounds).cgPath
        border.frame = addImgBtn.bounds
        border.lineWidth = 1.0
        border.lineCap = CAShapeLayerLineCap(rawValue: "square")
        border.lineDashPattern = [2, 2]
        addImgBtn.layer.addSublayer(border)
        
        self.contentView.addSubview(addImgBtn)
        
    }
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
}
